playbooks/roles/edx_django_service/templates/edx/app/nginx/sites-available/discovery.j2

```diff
@@ -0,0 +1,104 @@
+#
+# {{ ansible_managed }}
+#
+
+{% if nginx_default_sites is defined and edx_django_service_name in nginx_default_sites %}
+  {% set default_site = "default_server" %}
+{% else %}
+  {% set default_site = "" %}
+{% endif %}
+
+upstream {{ edx_django_service_name }}_app_server {
+{% for host in nginx_edx_django_service_gunicorn_hosts %}
+    server {{ host }}:{{ edx_django_service_gunicorn_port }} fail_timeout=0;
+{% endfor %}
+}
+
+# The Origin request header indicates where a fetch originates from. It doesn't include any path information,
+# but only the server name (e.g. https://www.example.com).
+# See https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Origin for details.
+#
+# Here we set the value that is included in the Access-Control-Allow-Origin response header. If the origin is one
+# of our known hosts--served via HTTP or HTTPS--we allow for CORS. Otherwise, we set the "null" value, disallowing CORS.
+map $http_origin $cors_origin {
+default "null";
+{% for host in edx_django_service_cors_whitelist %}
+  "~*^https?:\/\/{{ host|replace('.', '\.') }}$" $http_origin;
+{% endfor %}
+}
+
+server {
+  server_name {{ DISCOVERY_HOSTNAME }};
```

Comment:
What about setting the variable `discovery_service_name`, I guess that should work instead of copying this file. Check those files out:

1. `discovery_service_name: "discovery"`:
  https://github.com/edx/configuration/blob/134ee868dd19461ae610127bb511ce61ff18b3f3/playbooks/roles/discovery/defaults/main.yml#L18

2. `edx_django_service_name: '{{ discovery_service_name }}'`:
  https://github.com/edx/configuration/blob/134ee868dd19461ae610127bb511ce61ff18b3f3/playbooks/roles/discovery/meta/main.yml#L25

3. `edx_django_service_hostname: '~^((stage|prod)-)?{{ edx_django_service_name }}.*'`:
  https://github.com/edx/configuration/blob/58f3de6777e45fbce0d476f52ec9e1b2b7d6797d/playbooks/roles/edx_django_service/defaults/main.yml#L67